// osrs-json-hiscores/lib/utils/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.rsnFromElement = exports.numberFromElement = exports.getActivityPageURL = exports.getSkillPageURL = exports.getPlayerTableURL = exports.getStatsURL = void 0;
var constants_1 = require("./constants");
exports.getStatsURL = function (gamemode, rsn) {
    return "" + constants_1.GAMEMODE_URL[gamemode] + constants_1.STATS_URL + encodeURIComponent(rsn);
};
exports.getPlayerTableURL = function (gamemode, rsn) {
    return "" + constants_1.GAMEMODE_URL[gamemode] + constants_1.SCORES_URL + "table=0&user=" + encodeURIComponent(rsn);
};
exports.getSkillPageURL = function (gamemode, skill, page) {
    return "" + constants_1.GAMEMODE_URL[gamemode] + constants_1.SCORES_URL + "table=" + constants_1.SKILLS.indexOf(skill) + "&page=" + page;
};
exports.getActivityPageURL = function (gamemode, activity, page) {
    return "" + constants_1.GAMEMODE_URL[gamemode] + constants_1.SCORES_URL + "category_type=1&table=" + constants_1.ACTIVITIES.indexOf(activity) + "&page=" + page;
};
exports.numberFromElement = function (el) {
    var _a;
    var innerHTML = (el || {}).innerHTML;
    var number = (_a = innerHTML === null || innerHTML === void 0 ? void 0 : innerHTML.replace(/[\n|,]/g, '')) !== null && _a !== void 0 ? _a : '-1';
    return parseInt(number, 10);
};
exports.rsnFromElement = function (el) {
    var innerHTML = (el || {}).innerHTML;
    return (innerHTML === null || innerHTML === void 0 ? void 0 : innerHTML.replace(/\uFFFD/g, ' ')) || '';
};
